### ISSUE TITLE
GeneratePolymorphicSchemas to include none abstract base type

### ISSUE THREAD

**Issue by MohammadYounes** [domaindrivendev/Swashbuckle.AspNetCore#1433]
I'm using the new "GeneratePolymorphicSchemas" feature in v5.0.0-rc6-preview-1090.

Once enabled, the schema was updated, listing all sub types excluding the base type. In my case, the base type is not abstract and is a valid input by itself.

I tried using my own `subTypesResolver`, but once I include the base type it goes into infinite loop.

Would you consider adding this ?

**Comment by xlegalles** [comment]
I confirm: just use your default implementation as a resolver and you get a `StackOverfowException`

```
c.SchemaGeneratorOptions.SubTypesResolver = SubTypesResolver;
private IEnumerable<Type> SubTypesResolver(Type baseType)
    {
      if (baseType == typeof (object))
        return Enumerable.Empty<Type>();
      return ((IEnumerable<Type>) baseType.Assembly.GetTypes()).Where<Type>((Func<Type, bool>) (type => type.IsSubclassOf(baseType)));
    }
```

**Comment by mix4242** [comment]
Can I just confirm then, for a C# Model like this (Using `GeneratePolymorphicSchemas`)
```
    public class SomeClass
    {
        public SecondLevelModel UniqueModelName { get; set; }
    }

    public class FirstLevelModel
    {
        public string Level1 { get; set; }
    }

    public class SecondLevelModel : FirstLevelModel
    {
        public string Level2 { get; set; }
    }

    public class ThirdLevelModel : SecondLevelModel
    {
        public string Level3 { get; set; }
    }
```

The expected output should NOT be:
```
"uniqueModelName": {
   "oneOf": [
      {
         "$ref": "#/components/schemas/ThirdLevelModel"
      }
   ],
   "nullable": true
}
```

But rather: (this is what I get when I deleted the `ThirdLevelModel`):
```
"uniqueModelName": {
   "$ref": "#/components/schemas/SecondLevelModel"
}
```

P.S. Shouldn't the nullable be present when I deleted the `ThirdLevelModel`?

Thanks!

**Comment by domaindrivendev** [comment]
As of version `5.6.0`, Inheritance & Polymorphism support has been greatly improved and I believe your issue should now be addressed. That is, when polymorphism is enabled, non-abstract base types will additionally be listed in the `oneOf` array.

See https://github.com/domaindrivendev/Swashbuckle.AspNetCore#inheritance-and-polymorphism for more info on the new features and config options.

**Comment by csMACnz** [comment]
I just updated to this package, I was relying on this behaviour before.

Case:
Class A is subType 1
Class B is subType 2
Class C is a superset (union) of both types so that all properties deserialise, and then can manually parse the difference.

There is no inheritance here, but i used to be able to use the subtype system so that swagger documentation said it is either an A or B, but the serialiser treats it as a C (clearer intent, easier implementation)

Any chance of allowing configuration to decide if we document non-abstract base types instead of forcing this breaking change?

(I've worked around this using Schema Filters but it isn't ideal)
